frontend/components/Layout.tsx

```diff
@@ -1,12 +1,26 @@
 import Header from './Header';
 import Footer from './Footer';
 import { useEffect, useLayoutEffect } from 'react';
-
+import {motion} from 'framer-motion';
 const Layout = ({ children }: any) => {
+
+  const variants = {
+    hidden: { opacity: 0, x: -200, y: 0 },
+    enter: { opacity: 1, x: 0, y: 0 },
+    exit: { opacity: 0, x: 0, y: -100 },
+}
   return (
-    <div className='flex min-h-screen flex-col gap-20 px-5 py-5 dark:bg-black md:px-20'>
+    <div className='flex flex-col min-h-screen gap-20 px-5 py-5 md:px-20 bg-primary dark:bg-primary-dark'>
       <Header />
-      <main className='flex w-full grow justify-center'>{children}</main>
+      <motion.main className='flex w-full grow justify-center'
+        variants={variants} // Pass the variant object into Framer Motion 
+        initial="hidden" // Set the initial state to variants.hidden
+        animate="enter" // Animated state to variants.enter
+        exit="exit" // Exit state (used later) to variants.exit
+        transition={{ type: 'linear' }} // Set the transition to linear
```

Comment:
Some comments here are not really required. Might want to remove all comments other than line 18, 19